src/Factory.php

```diff
@@ -68,18 +69,43 @@ public function add(string $set, array $options): self
     public function registerComponents(): void
     {
         foreach ($this->sets as $set) {
-            foreach ($this->filesystem->allFiles($set['path']) as $file) {
-                $path = array_filter(explode('/', Str::after($file->getPath(), $set['path'])));
-
-                Blade::component(
-                    SvgComponent::class,
-                    implode('.', array_filter($path + [$file->getFilenameWithoutExtension()])),
-                    $set['prefix']
-                );
+            if (isset($set['path'])) {
+                $this->registerComponentsByPath($set);
+            } else {
+                $this->registerComponentsByDisk($set);
             }
         }
     }
 
+    protected function registerComponentsByPath($set) : void
+    {
+        foreach ($this->filesystem->allFiles($set['path']) as $file) {
+            $relativePath = Str::after($file->getPath(), $set['path']);
+            $relativePath = str_replace(DIRECTORY_SEPARATOR, '/', $relativePath);
```

Comment:
Why did you change this?

Reply:
I am using windows.
Subfolder dot notation like <x-icon-solid.camera/> is not work in my side.
This code fixed it.